src/reactviews/pages/QueryResult/queryResultPane.tsx

```diff
@@ -285,7 +285,9 @@ export const QueryResultPane = () => {
 
     const renderGridPanel = () => {
         const grids = [];
-        gridRefs.current.forEach((r) => r?.refreshGrid());
+        if (!metadata?.isExecutionPlan) {
+            gridRefs.current.forEach((r) => r?.refreshGrid());
```

Comment:
Also the correct fix will be to prevent refreshing the grid all the times. IIUC, this just means that the grid is not refreshed repeatedly when the query plan is present. This won't fix the issue when query plan isn't present, right? 

@caohai , what do you think?

Reply:
Any idea why the query plan being refreshed constantly resulted in flickering, but a regular result set didn't flicker?  (not so say it didn't also have the issue you're investigating, but I'm curious why it manifested differently)

Reply:
IIRC the repeated refreshing issue only happens when the query plan is present.

Reply:
execution plans only load after reading the resulting xml showplan of the query. therefore, it updates the state once the results are loaded, which causes a rendering loop if the grid gets refreshed